clientv3/integration/watch_test.go

```
@@ -868,3 +868,39 @@ func TestWatchCancelAndCloseClient(t *testing.T) {
 	<-donec
 	clus.TakeClient(0)
 }
+
+func TestWatchPrevKV(t *testing.T) {
+	cluster := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer cluster.Terminate(t)
+	etcdcli := cluster.RandClient()
+
+	tests := []struct {
+		key  string
+		vals []string
+	}{{
+		key:  "foo",
+		vals: []string{"bar1", "bar2"},
+	}, {
+		key:  "/abc",
+		vals: []string{"first", "second"},
+	}}
+	for i, tt := range tests {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+
+		etcdcli.Put(ctx, tt.key, tt.vals[0])
+		wch := etcdcli.Watch(ctx, tt.key, clientv3.WithPrefix(), clientv3.WithPrevKV())
+		etcdcli.Put(ctx, tt.key, tt.vals[1])
+
+		select {
+		case watchResp := <-wch:
+			if tt.vals[1] != string(watchResp.Events[0].Kv.Value) {
+				t.Errorf("#%d: unequal value: want=%s, get=%s", i, tt.vals[1], watchResp.Events[0].Kv.Value)
+			}
+			if tt.vals[0] != string(watchResp.Events[0].PrevKv.Value) {
+				t.Errorf("#%d: unequal value: want=%s, get=%s", i, tt.vals[0], watchResp.Events[0].PrevKv.Value)
+			}
+		case <-ctx.Done():
```

this should raise a timeout error